### ISSUE TITLE
session.get breaks up certain strings in params

### ISSUE THREAD

**Issue by isparavanje** [aio-libs/aiohttp#2146]
## Long story short

When strings in the params contains semicolons, the string is not parsed into the URL correctly.

## Expected behaviour

with `params={'mode':'fastest;car;traffic:enabled'}`, one expects the url to be `...?mode=fastest;car;traffic:enabled`

## Actual behaviour

The url from response.url is `...?mode=fastest&car=&traffic:enabled=`

## Steps to reproduce
```
async with session.get(url ,params={'mode':'fastest;car;traffic:enabled'}) as response:
        print(response.url)
```
## Your environment

Windows 10, Anaconda 4.4.0, Python 3.5.3, jupyter 4.3.0
This issue was encountered with the HERE routing API.

**Comment by asvetlov** [comment]
Did you mean `params={...}`?

**Comment by asvetlov** [comment]
In former w3c recommendation `;` is processed exactly as `&`.
Python's stdlib does the same BTW

See also https://github.com/aio-libs/aiohttp/issues/1901

**Comment by isparavanje** [comment]
Yes, I meant params, I will change that. I do not know about the w3c recommendation, but it is notable that some well known rest APIs deviate from that behaviour (HERE), and that the requests module behaves as expected with semicolons. However, if you believe that it is not a bug then you could close this issue as the workaround is rather trivial.

**Comment by argaen** [comment]
@asvetlov didn't https://github.com/aio-libs/yarl/pull/104 solve this? Maybe we can close

**Comment by asvetlov** [comment]
I not 100% sure about aio-libs/yarl#104 correctness.

It behaves like requests which is good.
But on other hand sometimes quoting is not desirable, we need special boolean flag in `with_query()`.
`.with_query(str)` behavior also should be reviewed.

Let's keep the issue open, I cope with it (and other URL related ones) in a few days.

**Comment by asvetlov** [comment]
Fixed by `yarl==0.13.0`
The implementation encodes `;` with `%3B` though.
